tools/monorepo-utils/src/workflow-profiler/lib/github.ts

```diff
@@ -0,0 +1,55 @@
+/**
+ * Internal dependencies
+ */
+import { octokitWithAuth } from '../../core/github/api';
+
+export type PaginatedDataTotals = {
```

Comment:
The usage of the words "data", "total", "count", "total_count", etc. feels rather confusing here, and I'd love to help create some more clarity here. Could you give me a quick overview of what happens here, what we're fetching, and so on? With more clarity around this I hope we'd be able to reword things here and in `lib/index.ts` to be more comprehensible by someone not super familiar with the specific API calls and their formats.

Reply:
I updated to this for some more clarity. This function is used to process multiple pages of GitHub data by keeping track of running totals. The only requirements are `count` and `total_number` in this object. A processing function is also passed into `requestPaginatedData` to handle each page's data by updating the `PaginatedDataTotals` object.

```js
export type PaginatedDataTotals = {
	// count is the running total of items processed
	count: number;
	// total_count is the total number of items available
	total_count: number;
	// Any other data that needs to be tracked
	// eslint-disable-next-line @typescript-eslint/no-explicit-any
	[ key: string ]: any;
};
```